### PR TITLE
Apply Azure Container Instance API 2017-10-01-preview

### PR DESCRIPTION
**Pull request by yolocs**
## Description

New spec: https://github.com/Azure/azure-rest-api-specs/pull/1863
Changes:
* Added container run-to-completion option
* Added support for azure file volume mount
* Some model-wise changes

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
You are missing description about this PR as well as link to the spec PR.
Also when you use generate.cmd to regenerate your SDK, you will get a new meta data file .txt (that is also missing)
please address all these and this is good to merge.

**Comment by yolo3301** [comment]
@shahabhijeet I think the PR title is informative enough? Or you want more details like what's actually changed in the new API version? Also the new spec link **is** already in the description section.

**Comment by yolo3301** [comment]
@shahabhijeet Addressed the comments.

File: src/SDKs/ContainerInstance/Management.ContainerInstance/Microsoft.Azure.Management.ContainerInstance.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
   <PropertyGroup>
     <Description>Microsoft Azure Container Instance Management Library</Description>
-    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <VersionPrefix>1.0.1-preview</VersionPrefix>
```
Comment (shahabhijeet) [inline src/SDKs/ContainerInstance/Management.ContainerInstance/Microsoft.Azure.Management.ContainerInstance.csproj]: @yolo3301 missing package release notes like below
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/BatchAI/Management.BatchAI/Microsoft.Azure.Management.BatchAI.csproj#L10

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
If you have merged the spec, can you provide the PR where you made the spec changes.
I need the PR link, that gives me an idea the feedback you received on your spec changes and gives me an idea about the sdk changes resulted due to it.

File: src/SDKs/ContainerInstance/Management.ContainerInstance/Microsoft.Azure.Management.ContainerInstance.csproj (hunk 1 above)
Comment (yolo3301) [reply src/SDKs/ContainerInstance/Management.ContainerInstance/Microsoft.Azure.Management.ContainerInstance.csproj]: @shahabhijeet the spec PR was already in the description. Pasting here again: Azure/azure-sdk-for-python#1557

Comment (yolo3301) [reply src/SDKs/ContainerInstance/Management.ContainerInstance/Microsoft.Azure.Management.ContainerInstance.csproj]: @shahabhijeet Have been using the wrong PR. The right one is: https://github.com/Azure/azure-rest-api-specs/pull/1863  Also updated the one in the description.

**Comment by yolo3301** [comment]
@shahabhijeet the build failure doesn't look related with my latest change. Can you verify?

**Pull request merged**